### PR TITLE
Edit button

### PR DESCRIPTION
**Pull request by marcterre**
I added a Form and a counter component, they are both the same than before. You could ignore this parts. Its all about my "ModalEdit" component and what comes with it. ✌️

### CODE REVIEW

File: pages/_app.js [hunk 1]
```diff
@@ -50,6 +50,32 @@ export default function App({ Component, pageProps }) {
...
+  function handleViewDetailsChange(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    setViews(
+      views.map((entry) =>
+        entry.id === currentView.id ? { ...entry, ...data } : entry
+      )
```
Comment (F-Kirchhoff) [inline pages/_app.js:75]: I have a question. Is a view independent from a project or is it strictly tied to it. If the latter one is true you have two options:
1. the views need to moved into the data structure of your project objects. That means that there should be a new key value pair in each project object that is called "views". The value is an array of your view objects that is linked to that project.
2. You create an independent views list like you did here. but each project still needs an "views" key which is a list of __view IDs__. If updating your views you need to update the views list as well as the respective project (if you add or remove a view from the project). 

It is your choice which way to go, I guess the first one is easier at the start.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Marc, you are already using components to reuse code, thats cool. At some points I guess you introduce too much complexity though. Have a look at my comments. About your views: lets have a look at your data structure if you are not sure which path to choose.

File: pages/_app.js (hunk 1 above)
Comment (marcterre) [reply pages/_app.js:75]: They are independet from each other 👍

File: components/Form.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  const [countDescription, setCountDescription] = useState(0);
+
+  return (
+    <StyledForm onSubmit={handleSubmit}>
+      <label htmlFor="name">Name:</label>
+      <input
+        id="name"
+        name="name"
+        type="text"
+        pattern="^[a-zA-Z0-9äüöÄÜÖ][a-zA-Z0-9-_ äüöÄÜÖß.] {1,}"
```
Comment (niklasrammelkamp) [inline components/Form.js:16]: Hi Marc, somehow I have the feeling that this pattern is preventing that I can type any Name. I checked the preview just to remember what your app is about and I can type any Name.

**Review by niklasrammelkamp — COMMENTED** [review]
Hi Marc, Looks good to me I only have one question.

File: components/Form.js (hunk 2 above)
Comment (marcterre) [reply components/Form.js]: Hey niklas, i think the problem was the space after the second []

Comment (AstridOttmann) [reply components/Form.js]: Hey Marc, it works like it is supposed to. 
Just one note: 
if you use the English language you should allow apostrophes (') in your pattern.

Comment (marcterre) [reply components/Form.js]: Thank you, the pattern is still being improved. I will implement this in the next pr 🙂

**Pull request merged**